sqlmesh/integrations/github/cicd/controller.py

```diff
@@ -0,0 +1,680 @@
+                        h("th", "Change Type"),
+                        h("th", "Dates Loaded"),
+                    ],
+                ]
+                body_rows: List[Element | List[Element]] = []
+                for affected_model in pr_affected_models:
+                    model_rows = [
+                        h("td", affected_model.model_name),
+                        h("td", affected_model.change_category_str),
+                    ]
+                    if affected_model.intervals:
+                        model_rows.append(h("td", affected_model.formatted_loaded_intervals))
+                    else:
+                        model_rows.append(h("td", "N/A"))
+                    body_rows.append(model_rows)
+                table_header = h("thead", [h("tr", row) for row in header_rows])
+                table_body = h("tbody", [h("tr", row) for row in body_rows])
+                summary = str(h("table", [table_header, table_body]))
+            self._update_check(
+                name="SQLMesh - PR Environment Synced",
+                status=status,
+                conclusion=conclusion,
+                title=title,
+                summary=summary,
+            )
+            self.update_sqlmesh_comment_info(
+                value=f"- PR Virtual Data Environment: `{self.pr_environment_name}`",
+                find_regex=r"- PR Virtual Data Environment: `.*`",
+                replace_if_exists=False,
+            )
+        else:
+            conclusion_to_summary = {
+                GithubCheckConclusion.SKIPPED: f":next_track_button: Skipped creating or updating PR Environment `{self.pr_environment_name}` since a prior stage failed",
+                GithubCheckConclusion.FAILURE: f":x: Failed to create or update PR Environment `{self.pr_environment_name}`. There are likely uncateogrized changes. Run `plan` to apply these changes.",
+                GithubCheckConclusion.CANCELLED: f":stop_sign: Cancelled creating or updating PR Environment `{self.pr_environment_name}`",
+                GithubCheckConclusion.ACTION_REQUIRED: f":warning: Action Required to create or update PR Environment `{self.pr_environment_name}`. There are likely uncateogrized changes. Run `plan` to apply these changes.",
+            }
+            summary = conclusion_to_summary.get(
+                conclusion, f":interrobang: Got an unexpected conclusion: {conclusion.value}"
+            )
+            self._update_check(
+                name="SQLMesh - PR Environment Synced",
+                status=status,
+                conclusion=conclusion,
+                title=title,
+                summary=summary,
+            )
+
+    def update_prod_environment_check(
+        self, status: GithubCheckStatus, conclusion: t.Optional[GithubCheckConclusion] = None
+    ) -> None:
+        """
+        Updates the status of the merge commit for the prod environment.
+        """
+        status_to_title = {
```

Comment:
constants

Reply:
These are localized constants to these functions. I would prefer to keep them where they are.

Reply:
i don't see why that matters, this dict has to be created every time. let's move it to the class level constant

Reply:
For example `status_to_title` exists in 4 different functions for 4 different checks. We would need to move these up to the top of the class and then prefix them all saying which one is for which of these classes. Then when someone is reading they would need to check the top of the class to see what the definition is. We would have done this to avoid create 4 different dicts of 2 keys each on each execution.

Reply:
ah you have 4 different status to titles?

Reply:
i see let's dry it up and make it a function, didn't realize you had duplicated it 

is it possible to dry this up and pass in this dictionary to the function?

Reply:
Yeah one for each check we do. Their mapping changes based on the check being performed.

Reply:
Let me see